akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1013,11 +1012,7 @@ trait FlowOps[+Out, +Mat] {
    * '''Cancels when''' downstream cancels
    *
    */
-  def flatten[U](strategy: FlattenStrategy[Out, U]): Repr[U, Mat] = strategy match {
-    case scaladsl.FlattenStrategy.Concat | javadsl.FlattenStrategy.Concat ⇒ deprecatedAndThen(ConcatAll())
-    case _ ⇒
-      throw new IllegalArgumentException(s"Unsupported flattening strategy [${strategy.getClass.getName}]")
-  }
+  def flattenConcat[U]()(implicit ev: Out <:< Source[U, _]): Repr[U, Mat] = deprecatedAndThen(ConcatAll())
```

Comment:
Who is the dog?

-- 
Cheers,
√
On 3 Nov 2015 11:23, "Konrad Malawski" <notifications@github.com> wrote:

> In akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala
> <https://github.com/akka/akka/pull/18809#discussion_r43732681>:
>
> > @@ -1013,11 +1012,7 @@ trait FlowOps[+Out, +Mat] {
> >     * '''Cancels when''' downstream cancels
> >     *
> >     */
> > -  def flatten[U](strategy: FlattenStrategy[Out, U]): Repr[U, Mat] = strategy match {
> > -    case scaladsl.FlattenStrategy.Concat | javadsl.FlattenStrategy.Concat ⇒ deprecatedAndThen(ConcatAll())
> > -    case _ ⇒
> > -      throw new IllegalArgumentException(s"Unsupported flattening strategy [${strategy.getClass.getName}]")
> > -  }
> > +  def flattenConcat[U]()(implicit ev: Out <:< Source[U, _]): Repr[U, Mat] = deprecatedAndThen(ConcatAll())
>
> Oh in scaladsl as well. Ok I guess, makes the APIs consistent.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/18809/files#r43732681>.
>